fabric_cf/actor/fim/plugins/broker/aggregate_bqm_plugin.py

```diff
@@ -34,20 +35,70 @@
+class AggregatedBQMPlugin:
     """
     Implement a plugin for simple aggregation of CBM into BQM, transforming site
     topologies into CompositeNodes and linking them with (Composite)Links
     This is based on fim.pluggable BrokerPluggable
     """
+    # set to true to test creating ABQM without talking to database or requiring
+    # an actor reference
+    DEBUG_FLAG = False
 
     def __init__(self, actor, logger=None):
+        if not self.DEBUG_FLAG:
+            assert actor is not None
         self.actor = actor
-        self.log = logger
+        self.logger = logger
+
+    def __occupied_node_capacity(self, *, node_id: str) -> Tuple[Capacities,
+                                                                 Dict[ComponentType, Dict[str, Capacities]]]:
+        """
+        Figure out the total capacity occupied in the network node and return a tuple of
+        capacities occupied in this node and a dict of component capacities that are occupied
+        organized by component type and model.
+        """
+        assert node_id is not None
+        # get existing reservations for this node
+        existing_reservations = self.actor.get_plugin().get_database().\
+            get_reservations_by_graph_node_id(graph_node_id=node_id)
+
+        # node capacities
+        occupied_capacities = Capacities()
+        occupied_component_capacities = defaultdict(dict)
+        # Remove allocated capacities to the reservations
+        if existing_reservations is not None:
+            for reservation in existing_reservations:
+                # For Active or Ticketed or Ticketing reservations; compute the counts from available
+                allocated_sliver = None
+                if reservation.is_ticketing() and reservation.get_approved_resources() is not None:
+                    allocated_sliver = reservation.get_approved_resources().get_sliver()
+
+                if (reservation.is_active() or reservation.is_ticketed()) and \
+                        reservation.get_resources() is not None:
+                    allocated_sliver = reservation.get_resources().get_sliver()
+
+                if allocated_sliver is not None:
+                    occupied_capacities = occupied_capacities + allocated_sliver.get_capacities()
+
+                    if allocated_sliver.attached_components_info is not None:
+                        for allocated_component in allocated_sliver.attached_components_info.devices.values():
+                            rt = allocated_component.resource_type
+                            rm = allocated_component.resource_model
+                            if occupied_component_capacities[rt].get(rm, None) is None:
+                                occupied_component_capacities[rt][rm] = Capacities()
+
+                            occupied_component_capacities[rt][rm] = occupied_component_capacities[rt][rm] + \
+                                                          allocated_component.capacities
```

Comment:
allocated_component.capacities should be changed to allocated_component.capacity_allocations

Reply:
This depends on how we use the fields. If we already use capacity_allocations, then you are correct

Reply:
I fixed this in the following commit